### PR TITLE
ShowMeshBoundingBox

### PR DESCRIPTION
**Pull request by RubioJr9**
Added ShowMeshBoundingBox module and test network.

Tickets:
#1275
#1746

### CODE REVIEW

File: src/Modules/Visualization/ShowMeshBoundingBox.h [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    INPUT_PORT(0, InputField, Field);
+    OUTPUT_PORT(0, OutputGeom, GeometryObject);
+
+    MODULE_TRAITS_AND_INFO(ModuleHasUI)
+
+    static const Core::Algorithms::AlgorithmParameterName XSize;
+    static const Core::Algorithms::AlgorithmParameterName YSize;
+    static const Core::Algorithms::AlgorithmParameterName ZSize;
+  private:
+    ShowMeshBoundingBoxImpl impl_;
```
Comment (dcwhite) [inline src/Modules/Visualization/ShowMeshBoundingBox.h]: Convert this to https://en.cppreference.com/w/cpp/language/pimpl with forward declaration of the impl class.

File: src/Modules/Visualization/ShowMeshBoundingBox.cc [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+  RenderState renState;
+  renState.set(RenderState::USE_NORMALS, false);
+  renState.set(RenderState::IS_ON, true);
+  renState.set(RenderState::USE_TRANSPARENCY, false);
+  renState.mGlyphType = RenderState::GlyphType::LINE_GLYPH;
+  renState.set(RenderState::USE_DEFAULT_COLOR, false);
+  return renState;
+}
+
+GeometryHandle ShowMeshBoundingBoxImpl::makeGeomemtry(const GeometryIDGenerator& idGen)
```
Comment (dcwhite) [inline src/Modules/Visualization/ShowMeshBoundingBox.cc]:
```suggestion
GeometryHandle ShowMeshBoundingBoxImpl::makeGeometry(const GeometryIDGenerator& idGen)
```

File: src/Modules/Visualization/ShowMeshBoundingBox.h (hunk 1 above)
Comment (dcwhite) [reply src/Modules/Visualization/ShowMeshBoundingBox.h]: Then the impl class resides totally in the cc file and compilation time is saved on future impl changes.

**Comment by dcwhite** [comment]
A more detailed explanation: https://herbsutter.com/gotw/_100/

There are several examples already in the module code, where an algorithm is not fully extracted yet. Just `ag "impl_->"`

File: src/Modules/Visualization/ShowMeshBoundingBox.h [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+#ifndef MODULES_VISUALIZATION_SHOW_MESH_BOUNDING_BOX_H
+#define MODULES_VISUALIZATION_SHOW_MESH_BOUNDING_BOX_H
+
+#include <Core/GeometryPrimitives/BBox.h>
+#include <Core/GeometryPrimitives/Point.h>
+#include <Core/GeometryPrimitives/Vector.h>
+#include <Dataflow/Network/GeometryGeneratingModule.h>
+#include <Graphics/Datatypes/GeometryImpl.h>
+#include <Graphics/Glyphs/GlyphGeom.h>
```
Comment (dcwhite) [inline src/Modules/Visualization/ShowMeshBoundingBox.h]: Almost all of these includes now belong in the cc file only, the goal of pimpl

**Comment by dcwhite** [comment]
boxes
Looks great!

**Comment by dcwhite** [comment]
@jessdtate @allywarner Ready to review.

**Comment by allywarner** [comment]
I also think it looks great! Why is the ShowAndEditDipole module part of the test network though?

**Comment by RubioJr9** [comment]
> I also think it looks great! Why is the ShowAndEditDipole module part of the test network though?

I did it so it'd be easy to test if the module updates when the upstream port changes. Do you want me to remove it?

**Pull request merged**